src/source/RazorEngine.Core/Templating/RazorEngineService.cs

```diff
@@ -48,6 +51,12 @@ internal RazorEngineService(ITemplateServiceConfiguration config)
             _config = config;
             //_core = new RazorEngineCore(config, this);
             _core_with_cache = new RazorEngineCoreWithCache(new ReadOnlyTemplateServiceConfiguration(config), this);
+            AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolve;
+        }
+
+        private Assembly AssemblyResolve(object sender, ResolveEventArgs args)
+        {
+            return CompilerReference.Resolve(args.Name, _config.ReferenceResolver.GetReferences(null));
```

Comment:
This might be a bit dangerous. I think custom implementation of `IReferenceResolver` might not behave as expected when giving `null`. Ideally we would "save" the references once we retrieved them in the compilation process and them use the saved list for the `AssemblyResolve` event... But I think we could do that in a later PR. If you don't want to include it here please open an issue :).